### PR TITLE
feat: implement EIP-7742

### LINKED ISSUE #7176: Electra: Implement EIP7742: uncouple blob limits per block across CL and EL
### Problem description

As determined in [ACDC #144](https://github.com/ethereum/pm/issues/1178), EIP-7742 is moving into CFI with teams signalling to implement for Pectra devnet-5.

### Solution description

Reference: https://github.com/ethereum/consensus-specs/pull/3800
EIP: https://eips.ethereum.org/EIPS/eip-7742
Engine-API update: https://github.com/ethereum/execution-apis/pull/574 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ensi321**
Closes #7176

### CODE REVIEW

**Comment by philknows** [comment]
Included for alpha.10

File: packages/cli/src/options/beaconNodeOptions/chain.ts [hunk 1]
```diff
@@ -284,4 +288,20 @@ Will double processing times. Use only for debugging purposes.",
     default: defaultOptions.chain.maxCPStateEpochsInMemory,
     group: "chain",
   },
+
+  "chain.targetBlobsPerBlock": {
```
Comment (nflaig) [inline packages/cli/src/options/beaconNodeOptions/chain.ts]: these values will be part of spec config

Comment (ensi321) [reply packages/cli/src/options/beaconNodeOptions/chain.ts]: Setting 4/8 or 6/9 is part of the scope of EIP-7691. Will open a separate PR for it.

EIP-7742 only sets target = max / 2 and no new config value

File: packages/beacon-node/src/chain/produceBlock/produceBlockBody.ts [hunk 2]
```diff
@@ -586,6 +586,13 @@ function preparePayloadAttributes(
     (payloadAttributes as deneb.SSEPayloadAttributes["payloadAttributes"]).parentBeaconBlockRoot = parentBlockRoot;
   }
 
+  if (ForkSeq[fork] >= ForkSeq.electra) {
+    (payloadAttributes as electra.SSEPayloadAttributes["payloadAttributes"]).targetBlobsPerBlock = Math.floor(
+      MAX_BLOBS_PER_BLOCK / 2
+    );
+    (payloadAttributes as electra.SSEPayloadAttributes["payloadAttributes"]).maxBlobsPerBlock = MAX_BLOBS_PER_BLOCK;
```
Comment (twoeths) [inline packages/beacon-node/src/chain/produceBlock/produceBlockBody.ts:593]: cast to `electra.SSEPayloadAttributes` once

File: packages/beacon-node/src/execution/engine/http.ts [hunk 3]
```diff
@@ -201,7 +201,8 @@ export class ExecutionEngineHttp implements IExecutionEngine {
     executionPayload: ExecutionPayload,
     versionedHashes?: VersionedHashes,
     parentBlockRoot?: Root,
-    executionRequests?: ExecutionRequests
+    executionRequests?: ExecutionRequests,
+    maxBlobsPerBlock?: number
```
Comment (twoeths) [inline packages/beacon-node/src/execution/engine/http.ts:205]: `verifyBlocksExecutionPayload` passed in `targetBlobsPerBlock` which is correct as in the [api](https://github.com/ethereum/execution-apis/pull/574/files#diff-18471a795328cf06e000056d594a646c71fa9ce16c35511f3506543b6fc57981R83) but here it's `maxBlobsPerBlock`

**Comment by twoeths** [comment]
need to wait for [eip-7742](https://github.com/ethereum/execution-apis/pull/574) execution api to be merged

**Comment by ensi321** [comment]
Converting to draft as there is a blocker for the EIP.

**Comment by ensi321** [comment]
We've decided not to include EIP-7742 in Pectra

**Pull request closed without merging**